packages/web/src/dev/CopyToClipboard.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react'
+import { StyleSheet, Text, View } from 'react-native'
+import CopyIcon from 'src/icons/CopyIcon'
+import Hoverable from 'src/shared/Hoverable'
+import { colors } from 'src/styles'
+import { copyToClipboad } from 'src/utils/utils'
+
+interface Props {
+  content: string
+}
+
+interface State {
+  copied: boolean
+  hover: boolean
+}
+
+class CopyToClipboard extends React.PureComponent<Props, State> {
+  state = {
+    copied: false,
+    hover: false,
+  }
+
+  copy(event) {
+    event.preventDefault()
+    event.stopPropagation()
+    if (!this.state.copied) {
+      copyToClipboad(this.props.content)
+      this.setState({ copied: true })
+      setTimeout(() => this.setState({ copied: false }), 2000)
+    }
+  }
+
+  hover(hover: boolean) {
+    this.setState({ hover })
+  }
+
+  render() {
+    const { copied, hover } = this.state
+    const copy = this.copy.bind(this)
+    return (
+      <View style={[styles.container]} onClick={copy}>
+        {copied ? (
+          <Text style={[styles.copied]}>Copied</Text>
+        ) : (
+          <Text style={[styles.icon]}>
+            <Hoverable
+              onHoverIn={this.hover.bind(this, true)}
```

Comment:
rather than binding here would you create a class arrow function

```
hover = (hover: boolean) => {
    this.setState({ hover })
  }
```

binding within render creates a new function i believe which messes with prop  check heuristics, therefore we try to use on the above style 👍🏻

Reply:
What can I do with the functions that needs some params?